scripts/ASTImportTest.sh

```diff
@@ -14,90 +45,91 @@ SOLIDITY_BUILD_DIR=${SOLIDITY_BUILD_DIR:-${REPO_ROOT}/build}
+        # first failing test
+        # exit 1
+        FAILED=$((FAILED + 1))
+        printError "ERROR: AST reimport failed for input file $sol_file"
+        printError
+        printError "Compiler stderr:"
+        cat ./stderr.txt >&2
+        printError
+        printError "Compiler stdout:"
+        cat ./obtained.json >&2
+        return 1
+    fi
+    if ! diff_files expected.json obtained.json
+    then
+        FAILED=$((FAILED + 1))
+    fi
+    TESTED=$((TESTED + 1))
+    rm expected.json obtained.json
+    rm -f stderr.txt
+}
+
+# function tests whether exporting and importing again is equivalent.
 # Results are recorded by adding to FAILED or UNCOMPILABLE.
-# Also, in case of a mismatch a diff and the respective ASTs are printed
+# Also, in case of a mismatch a diff is printed
 # Expected parameters:
 # $1 name of the file to be exported and imported
 # $2 any files needed to do so that might be in parent directories
 function testImportExportEquivalence {
-    local nth_input_file="$1"
-    IFS=" " read -r -a all_input_files <<< "$2"
-
-    if $SOLC "$nth_input_file" "${all_input_files[@]}" > /dev/null 2>&1
+    local sol_file="$1"
+    local input_files="$2"
+    if "$SOLC" --bin "${input_files}" > ${REDIRECT_TO} 2>&1
     then
-        ! [[ -e stderr.txt ]] || { echo "stderr.txt already exists. Refusing to overwrite."; exit 1; }
-
-        # save exported json as expected result (silently)
-        $SOLC --combined-json ast --pretty-json "$nth_input_file" "${all_input_files[@]}" > expected.json 2> /dev/null
-        # import it, and export it again as obtained result (silently)
-        if ! $SOLC --import-ast --combined-json ast --pretty-json expected.json > obtained.json 2> stderr.txt
-        then
-            # For investigating, use exit 1 here so the script stops at the
-            # first failing test
-            # exit 1
-            FAILED=$((FAILED + 1))
-            echo -e "ERROR: AST reimport failed for input file $nth_input_file"
-            echo
-            echo "Compiler stderr:"
-            cat ./stderr.txt
-            echo
-            echo "Compiler stdout:"
-            cat ./obtained.json
-            return 1
-        fi
-        DIFF="$(diff expected.json obtained.json)"
-        if [ "$DIFF" != "" ]
-        then
-            if [ "$DIFFVIEW" == "" ]
-            then
-                echo -e "ERROR: JSONS differ for $1: \n $DIFF \n"
-                echo "Expected:"
-                cat ./expected.json
-                echo "Obtained:"
-                cat ./obtained.json
-            else
-                # Use user supplied diff view binary
-                $DIFFVIEW expected.json obtained.json
-            fi
-            FAILED=$((FAILED + 1))
-            return 2
-        fi
-        TESTED=$((TESTED + 1))
-        rm expected.json obtained.json
-        rm -f stderr.txt
+        ! [[ -e stderr.txt ]] || fail "stderr.txt already exists. Refusing to overwrite."
```

Comment:
```suggestion
        [[ -e stderr.txt ]] && fail "stderr.txt already exists. Refusing to overwrite."
```

Reply:
Ok, I just removed this check, because the tests are always executed in their own temporary test directory. So this should be always empty.